Sources/AudioKit/Nodes/Playback/AudioPlayer/AudioPlayer.swift

```diff
@@ -40,15 +50,30 @@ public class AudioPlayer: Node {
         }
     }
 
-    public var duration: TimeInterval {
-        bufferDuration ?? file?.duration ?? 0
+    /// Used to get the correct current time, after seeking
+    private var elapsedTimeOffset: Double = 0
+    public var currentTime: TimeInterval {
+        if isPlaying {
+            if let nodeTime = playerNode.lastRenderTime,
+               let playerTime = playerNode.playerTime(forNodeTime: nodeTime) {
+                return Double(elapsedTimeOffset) + (Double(playerTime.sampleTime) / playerTime.sampleRate)
+            }
+        }
+        return 0
```

Comment:
Consistency: 

If the player is paused, Its currentTime can be a positive non-zero number. 

If the code you've written for the "isPlaying" block is unreliable when the player is paused, consider storing the currentTime in an internal temp variable in the pause() method before actually pausing the player.